### PR TITLE
Ba meyer master

### PR DESCRIPTION
**Pull request by lqwert**
Hallo,
Merge Request wegen des Code-Styles, Pattern-Atlas, ...

ich hab den master auf meinem branch gemerged und scheint zu funktionieren. Falls es zu Problemen kommen sollte bitte mir schreiben, ich hoffe nicht :)

Viele Grüße
Luca

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -7,5 +9,10 @@ spring.jpa.hibernate.ddl-auto=update
 logging.level.com.patternpedia.api=debug
 spring.datasource.initialization-mode=always
 spring.jpa.properties.hibernate.enable_lazy_load_no_trans=true
+
+spring.security.user.password=pass
+spring.security.user.name=user
+
```
Comment (manuwei) [inline src/main/resources/application.properties:15]: can we store this sensitive information somewhere else?

File: src/main/resources/public.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzWsrJMl/TJHVVUztgLDJ
+E1zIXbqqE52sTyI0xUX0iQxCp5NckwVJXjQrhZ0UcrIa0u6nC5Q83A/3oYpu13mt
+5nYhokRUkJythCxvJlMNa+41Q5EKrWc2DV5CZl1BuD6/yHShuPqtfxyApfzQgdoj
+JsrPydZJxWDZYwvolEWoTLGSyc1WZIX8WQX7muAPnpBD0pK6p8OPnONjaLtjeXvh
+k6O95qSbZJFx1val0ws2ruWyHymO47VxRWiAhqtdYasZQlK4gh9aqofXSoTKKOZF
+6Q6TvUppWcPmhvpIwSWugoCNjHxrPJtD8aIYP+YV2V2iPiCGyayXSKIz8XiHELye
+3QIDAQAB
+-----END PUBLIC KEY-----
```
Comment (manuwei) [inline src/main/resources/public.txt:9]: can we store this sensitive information somewhere else?

File: src/main/java/com/patternpedia/api/rest/controller/CandidateController.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    ) {
+        this.candidateService = candidateService;
+        this.patternLanguageService = patternLanguageService;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * GET Methods
+     */
+    @GetMapping(value = "/getAll")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/CandidateController.java:53]: stick to REST conventions: "/candidates/"

File: src/main/java/com/patternpedia/api/rest/controller/CandidateController.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        List<EntityModel<CandidateModel>> candidates = this.candidateService.getAllCandidates()
+                .stream()
+                .map(candidate -> new EntityModel<>(CandidateModel.from(candidate)))
+//                        getPatternViewLinks(patternView)))
+                .collect(Collectors.toList());
+
+        return new CollectionModel<>(candidates);
+    }
+
+    @GetMapping(value = "/getById/{candidateId}")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/CandidateController.java:65]: stick to REST conventions: "/candidates/{candidateId}"

File: src/main/java/com/patternpedia/api/rest/controller/CandidateController.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    @PreAuthorize(value = "#oauth2.hasScope('read')")
+    Candidate getCandidateById(@PathVariable UUID candidateId) {
+        return this.candidateService.getCandidateById(candidateId);
+    }
+
+    @GetMapping(value = "/getByUri/{candidateUri}")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/CandidateController.java:71]: stick to REST conventions: "/candidates/?uri=..."

File: src/main/java/com/patternpedia/api/rest/controller/CandidateController.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    Candidate getCandidateByUri(@PathVariable String candidateUri) {
+        return this.candidateService.getCandidateByURI(candidateUri);
+    }
+
+    /**
+     * CREATE Methods
+     */
+    @PostMapping(value = "/create")
+//    @PostMapping(value = "/create")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/CandidateController.java:80]: stick to REST conventions:  this should be a post on "/candidates/"

File: src/main/java/com/patternpedia/api/rest/controller/CandidateController.java [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+//    @PreAuthorize(value = "#oauth2.hasScope('write')")
+//    @CrossOrigin(exposedHeaders = "Location")
+    @ResponseStatus(HttpStatus.CREATED)
+    Candidate newCandidate(@RequestBody CandidateModel candidate) {
+        return this.candidateService.createCandidate(candidate);
+    }
+
+    @PostMapping(value = "/createComment/{candidateId}&{userId}")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/CandidateController.java:88]: stick to REST conventions: "/candidates/{candidateId/comments/{userId}" (userId can stay an URL param if you'd like to)

File: src/main/java/com/patternpedia/api/rest/controller/CandidateController.java [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+//        candidate.setName(issue.getName());
+//        candidate.setPatternLanguage(patternLanguageId.toString());
+////        return candidate;
+//        return this.candidateService.createCandidate(candidate);
+//    }
+
+    /**
+     * UPDATE Methods
+     */
+    @PutMapping(value = "/update")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/CandidateController.java:123]: stick to REST conventions: "/candidates/{candidateId}"

File: src/main/java/com/patternpedia/api/rest/controller/CandidateController.java [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+    @PutMapping(value = "/updateCommentRating/{candidateCommentId}&{userId}&{rating}")
+    Candidate putCandidateCommentRating(@PathVariable UUID candidateCommentId, @PathVariable UUID userId, @PathVariable String rating) {
+        return this.candidateService.commentUserRating(candidateCommentId, userId, rating);
+    }
+
+    /**
+     * DELETE Methods
+     */
+    @DeleteMapping(value = "/delete/{candidateId}")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/CandidateController.java:142]: stick to REST conventions: delete on "/candidates/{candidateId}"

File: src/main/java/com/patternpedia/api/rest/controller/CandidateController.java [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    /**
+     * UPDATE Methods
+     */
+    @PutMapping(value = "/update")
+    Candidate putCandidate(@RequestBody Candidate candidate) {
+        logger.info(candidate.toString());
+        return this.candidateService.updateCandidate(candidate);
+    }
+
+    @PutMapping(value = "/updateRating/{candidateId}&{userId}&{rating}")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/CandidateController.java:129]: stick to REST conventions: put on "/candidates/{candidateId}/rating/" - declare userId and rating as pathvariable, don't use '&' in the url mapping

File: src/main/java/com/patternpedia/api/rest/controller/IssueController.java [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+//@CrossOrigin(allowedHeaders = "*", origins = "*")
+@RequestMapping(value = "/issue", produces = "application/hal+json")
```
Comment (manuwei) [inline src/main/java/com/patternpedia/api/rest/controller/IssueController.java:19]: always in plural: issues/

**Comment by lqwert** [comment]
Hallo Manuela,
sorry das ich erst jetzt Antworte.
Danke für das viele Feedback, ich schau nochmal über meinen ganzen Code und schau das ich alles verbessere und würde dann ne neue PR stellen.
Ich hab die merge request gemacht ohne meinen Code groß nochmal anzugucken.

Viele Grüße
Luca



On 15. May 2020, at 09:33, manuwei <notifications@github.com<mailto:notifications@github.com>> wrote:



@manuwei commented on this pull request.

I haven't looked at everything get, but please have a look at standard REST naming conventions (https://restfulapi.net/resource-naming/), the DeepCode Analysis and get rid of all code that is commented out (unless there is a serious reason to keep it)

________________________________

In src/main/resources/application.properties<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425609583>:

> @@ -7,5 +9,10 @@ spring.jpa.hibernate.ddl-auto=update
 logging.level.com.patternpedia.api=debug
 spring.datasource.initialization-mode=always
 spring.jpa.properties.hibernate.enable_lazy_load_no_trans=true
+
+spring.security.user.password=pass
+spring.security.user.name=user
+


can we store this sensitive information somewhere else?

________________________________

In src/main/resources/public.txt<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425610058>:

> @@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzWsrJMl/TJHVVUztgLDJ
+E1zIXbqqE52sTyI0xUX0iQxCp5NckwVJXjQrhZ0UcrIa0u6nC5Q83A/3oYpu13mt
+5nYhokRUkJythCxvJlMNa+41Q5EKrWc2DV5CZl1BuD6/yHShuPqtfxyApfzQgdoj
+JsrPydZJxWDZYwvolEWoTLGSyc1WZIX8WQX7muAPnpBD0pK6p8OPnONjaLtjeXvh
+k6O95qSbZJFx1val0ws2ruWyHymO47VxRWiAhqtdYasZQlK4gh9aqofXSoTKKOZF
+6Q6TvUppWcPmhvpIwSWugoCNjHxrPJtD8aIYP+YV2V2iPiCGyayXSKIz8XiHELye
+3QIDAQAB
+-----END PUBLIC KEY-----


can we store this sensitive information somewhere else?

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425611258>:

> +    private ObjectMapper objectMapper;
+
+    public CandidateController(
+            CandidateService candidateService,
+            PatternLanguageService patternLanguageService,
+            ObjectMapper objectMapper
+    ) {
+        this.candidateService = candidateService;
+        this.patternLanguageService = patternLanguageService;
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * GET Methods
+     */
+    @GetMapping(value = "/getAll")


stick to REST conventions: "/candidates/"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425611367>:

> +    /**
+     * GET Methods
+     */
+    @GetMapping(value = "/getAll")
+    CollectionModel<EntityModel<CandidateModel>> all() {
+
+        List<EntityModel<CandidateModel>> candidates = this.candidateService.getAllCandidates()
+                .stream()
+                .map(candidate -> new EntityModel<>(CandidateModel.from(candidate)))
+//                        getPatternViewLinks(patternView)))
+                .collect(Collectors.toList());
+
+        return new CollectionModel<>(candidates);
+    }
+
+    @GetMapping(value = "/getById/{candidateId}")


stick to REST conventions: "/candidates/{candidateId}"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425612057>:

> +        List<EntityModel<CandidateModel>> candidates = this.candidateService.getAllCandidates()
+                .stream()
+                .map(candidate -> new EntityModel<>(CandidateModel.from(candidate)))
+//                        getPatternViewLinks(patternView)))
+                .collect(Collectors.toList());
+
+        return new CollectionModel<>(candidates);
+    }
+
+    @GetMapping(value = "/getById/{candidateId}")
+    @PreAuthorize(value = "#oauth2.hasScope('read')")
+    Candidate getCandidateById(@PathVariable UUID candidateId) {
+        return this.candidateService.getCandidateById(candidateId);
+    }
+
+    @GetMapping(value = "/getByUri/{candidateUri}")


stick to REST conventions: "/candidates/?uri=..."

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425612225>:

> +    @GetMapping(value = "/getById/{candidateId}")
+    @PreAuthorize(value = "#oauth2.hasScope('read')")
+    Candidate getCandidateById(@PathVariable UUID candidateId) {
+        return this.candidateService.getCandidateById(candidateId);
+    }
+
+    @GetMapping(value = "/getByUri/{candidateUri}")
+    Candidate getCandidateByUri(@PathVariable String candidateUri) {
+        return this.candidateService.getCandidateByURI(candidateUri);
+    }
+
+    /**
+     * CREATE Methods
+     */
+    @PostMapping(value = "/create")
+//    @PostMapping(value = "/create")


stick to REST conventions: this should be a post on "/candidates/"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425612712>:

> +        return this.candidateService.getCandidateByURI(candidateUri);
+    }
+
+    /**
+     * CREATE Methods
+     */
+    @PostMapping(value = "/create")
+//    @PostMapping(value = "/create")
+//    @PreAuthorize(value = "#oauth2.hasScope('write')")
+//    @CrossOrigin(exposedHeaders = "Location")
+    @ResponseStatus(HttpStatus.CREATED)
+    Candidate newCandidate(@RequestBody CandidateModel candidate) {
+        return this.candidateService.createCandidate(candidate);
+    }
+
+    @PostMapping(value = "/createComment/{candidateId}&{userId}")


stick to REST conventions: "/candidates/{candidateId/comments/{userId}" (userId can stay an URL param if you'd like to)

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425612958>:

> +//        JSONObject contentObject = new JSONObject();
+//        contentObject.put("content", contentArray);
+//
+//        Candidate candidate = new Candidate();
+//        candidate.setUri(issue.getName());
+//        candidate.setContent(contentObject);
+//        candidate.setName(issue.getName());
+//        candidate.setPatternLanguage(patternLanguageId.toString());
+////        return candidate;
+//        return this.candidateService.createCandidate(candidate);
+//    }
+
+    /**
+     * UPDATE Methods
+     */
+    @PutMapping(value = "/update")


stick to REST conventions: "/candidates/{candidateId}"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425613177>:

> +    }
+
+    @PutMapping(value = "/updateRating/{candidateId}&{userId}&{rating}")
+    Candidate putCandidateRating(@PathVariable UUID candidateId, @PathVariable UUID userId, @PathVariable String rating) {
+        return this.candidateService.userRating(candidateId, userId, rating);
+    }
+
+    @PutMapping(value = "/updateCommentRating/{candidateCommentId}&{userId}&{rating}")
+    Candidate putCandidateCommentRating(@PathVariable UUID candidateCommentId, @PathVariable UUID userId, @PathVariable String rating) {
+        return this.candidateService.commentUserRating(candidateCommentId, userId, rating);
+    }
+
+    /**
+     * DELETE Methods
+     */
+    @DeleteMapping(value = "/delete/{candidateId}")


stick to REST conventions: delete on "/candidates/{candidateId}"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425614184>:

> +//        candidate.setName(issue.getName());
+//        candidate.setPatternLanguage(patternLanguageId.toString());
+////        return candidate;
+//        return this.candidateService.createCandidate(candidate);
+//    }
+
+    /**
+     * UPDATE Methods
+     */
+    @PutMapping(value = "/update")
+    Candidate putCandidate(@RequestBody Candidate candidate) {
+        logger.info<http://logger.info>(candidate.toString());
+        return this.candidateService.updateCandidate(candidate);
+    }
+
+    @PutMapping(value = "/updateRating/{candidateId}&{userId}&{rating}")


stick to REST conventions: put on "/candidates/{candidateId}/rating/" - declare userId and rating as pathvariable, don't use '&' in the url mapping

________________________________

In src/main/java/com/patternpedia/api/rest/controller/IssueController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425614384>:

> +import com.patternpedia.api.entities.issue.IssueComment;
+import com.patternpedia.api.entities.issue.Issue;
+import com.patternpedia.api.service.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+//@CrossOrigin(allowedHeaders = "*", origins = "*")
+@RequestMapping(value = "/issue", produces = "application/hal+json")


always in plural: issues/

**Comment by manuwei** [comment]
Hallo Luca,

kein Problem, dafür machen ja Code Reviews.
Könntest du nochmal kurz über die Kommentare im Code in PR, DeepCode und Durchlauf des Builds schauen und mir rückmelden, wie lange du ungefähr brauchst, die Änderungen einzuarbeiten? Wie ich sehe, hast du ja schon damit angefangen und es ist schon deutlich weniger auskommentierter Code da😊

Viele Grüße,
Manuela
Von: lqwert <notifications@github.com>
Gesendet: Freitag, 15. Mai 2020 11:26
An: PatternPedia/pattern-pedia-views-backend <pattern-pedia-views-backend@noreply.github.com>
Cc: Weigold, Manuela <m.weigold@iaas.uni-stuttgart.de>; Mention <mention@noreply.github.com>
Betreff: Re: [PatternPedia/pattern-pedia-views-backend] Ba meyer master (#6)

Hallo Manuela,
sorry das ich erst jetzt Antworte.
Danke für das viele Feedback, ich schau nochmal über meinen ganzen Code und schau das ich alles verbessere und würde dann ne neue PR stellen.
Ich hab die merge request gemacht ohne meinen Code groß nochmal anzugucken.

Viele Grüße
Luca



On 15. May 2020, at 09:33, manuwei <notifications@github.com<mailto:notifications@github.com<mailto:notifications@github.com%3cmailto:notifications@github.com>>> wrote:



@manuwei commented on this pull request.

I haven't looked at everything get, but please have a look at standard REST naming conventions (https://restfulapi.net/resource-naming/), the DeepCode Analysis and get rid of all code that is commented out (unless there is a serious reason to keep it)

________________________________

In src/main/resources/application.properties<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425609583>:

> @@ -7,5 +9,10 @@ spring.jpa.hibernate.ddl-auto=update
logging.level.com.patternpedia.api=debug
spring.datasource.initialization-mode=always
spring.jpa.properties.hibernate.enable_lazy_load_no_trans=true
+
+spring.security.user.password=pass
+spring.security.user.name=user
+


can we store this sensitive information somewhere else?

________________________________

In src/main/resources/public.txt<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425610058>:

> @@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAzWsrJMl/TJHVVUztgLDJ
+E1zIXbqqE52sTyI0xUX0iQxCp5NckwVJXjQrhZ0UcrIa0u6nC5Q83A/3oYpu13mt
+5nYhokRUkJythCxvJlMNa+41Q5EKrWc2DV5CZl1BuD6/yHShuPqtfxyApfzQgdoj
+JsrPydZJxWDZYwvolEWoTLGSyc1WZIX8WQX7muAPnpBD0pK6p8OPnONjaLtjeXvh
+k6O95qSbZJFx1val0ws2ruWyHymO47VxRWiAhqtdYasZQlK4gh9aqofXSoTKKOZF
+6Q6TvUppWcPmhvpIwSWugoCNjHxrPJtD8aIYP+YV2V2iPiCGyayXSKIz8XiHELye
+3QIDAQAB
+-----END PUBLIC KEY-----


can we store this sensitive information somewhere else?

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425611258>:

> + private ObjectMapper objectMapper;
+
+ public CandidateController(
+ CandidateService candidateService,
+ PatternLanguageService patternLanguageService,
+ ObjectMapper objectMapper
+ ) {
+ this.candidateService = candidateService;
+ this.patternLanguageService = patternLanguageService;
+ this.objectMapper = objectMapper;
+ }
+
+ /**
+ * GET Methods
+ */
+ @GetMapping(value = "/getAll")


stick to REST conventions: "/candidates/"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425611367>:

> + /**
+ * GET Methods
+ */
+ @GetMapping(value = "/getAll")
+ CollectionModel<EntityModel<CandidateModel>> all() {
+
+ List<EntityModel<CandidateModel>> candidates = this.candidateService.getAllCandidates()
+ .stream()
+ .map(candidate -> new EntityModel<>(CandidateModel.from(candidate)))
+// getPatternViewLinks(patternView)))
+ .collect(Collectors.toList());
+
+ return new CollectionModel<>(candidates);
+ }
+
+ @GetMapping(value = "/getById/{candidateId}")


stick to REST conventions: "/candidates/{candidateId}"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425612057>:

> + List<EntityModel<CandidateModel>> candidates = this.candidateService.getAllCandidates()
+ .stream()
+ .map(candidate -> new EntityModel<>(CandidateModel.from(candidate)))
+// getPatternViewLinks(patternView)))
+ .collect(Collectors.toList());
+
+ return new CollectionModel<>(candidates);
+ }
+
+ @GetMapping(value = "/getById/{candidateId}")
+ @PreAuthorize(value = "#oauth2.hasScope('read')")
+ Candidate getCandidateById(@PathVariable UUID candidateId) {
+ return this.candidateService.getCandidateById(candidateId);
+ }
+
+ @GetMapping(value = "/getByUri/{candidateUri}")


stick to REST conventions: "/candidates/?uri=..."

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425612225>:

> + @GetMapping(value = "/getById/{candidateId}")
+ @PreAuthorize(value = "#oauth2.hasScope('read')")
+ Candidate getCandidateById(@PathVariable UUID candidateId) {
+ return this.candidateService.getCandidateById(candidateId);
+ }
+
+ @GetMapping(value = "/getByUri/{candidateUri}")
+ Candidate getCandidateByUri(@PathVariable String candidateUri) {
+ return this.candidateService.getCandidateByURI(candidateUri);
+ }
+
+ /**
+ * CREATE Methods
+ */
+ @PostMapping(value = "/create")
+// @PostMapping(value = "/create")


stick to REST conventions: this should be a post on "/candidates/"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425612712>:

> + return this.candidateService.getCandidateByURI(candidateUri);
+ }
+
+ /**
+ * CREATE Methods
+ */
+ @PostMapping(value = "/create")
+// @PostMapping(value = "/create")
+// @PreAuthorize(value = "#oauth2.hasScope('write')")
+// @CrossOrigin(exposedHeaders = "Location")
+ @ResponseStatus(HttpStatus.CREATED)
+ Candidate newCandidate(@RequestBody CandidateModel candidate) {
+ return this.candidateService.createCandidate(candidate);
+ }
+
+ @PostMapping(value = "/createComment/{candidateId}&{userId}")


stick to REST conventions: "/candidates/{candidateId/comments/{userId}" (userId can stay an URL param if you'd like to)

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425612958>:

> +// JSONObject contentObject = new JSONObject();
+// contentObject.put("content", contentArray);
+//
+// Candidate candidate = new Candidate();
+// candidate.setUri(issue.getName());
+// candidate.setContent(contentObject);
+// candidate.setName(issue.getName());
+// candidate.setPatternLanguage(patternLanguageId.toString());
+//// return candidate;
+// return this.candidateService.createCandidate(candidate);
+// }
+
+ /**
+ * UPDATE Methods
+ */
+ @PutMapping(value = "/update")


stick to REST conventions: "/candidates/{candidateId}"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425613177>:

> + }
+
+ @PutMapping(value = "/updateRating/{candidateId}&{userId}&{rating}")
+ Candidate putCandidateRating(@PathVariable UUID candidateId, @PathVariable UUID userId, @PathVariable String rating) {
+ return this.candidateService.userRating(candidateId, userId, rating);
+ }
+
+ @PutMapping(value = "/updateCommentRating/{candidateCommentId}&{userId}&{rating}")
+ Candidate putCandidateCommentRating(@PathVariable UUID candidateCommentId, @PathVariable UUID userId, @PathVariable String rating) {
+ return this.candidateService.commentUserRating(candidateCommentId, userId, rating);
+ }
+
+ /**
+ * DELETE Methods
+ */
+ @DeleteMapping(value = "/delete/{candidateId}")


stick to REST conventions: delete on "/candidates/{candidateId}"

________________________________

In src/main/java/com/patternpedia/api/rest/controller/CandidateController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425614184>:

> +// candidate.setName(issue.getName());
+// candidate.setPatternLanguage(patternLanguageId.toString());
+//// return candidate;
+// return this.candidateService.createCandidate(candidate);
+// }
+
+ /**
+ * UPDATE Methods
+ */
+ @PutMapping(value = "/update")
+ Candidate putCandidate(@RequestBody Candidate candidate) {
+ logger.info<http://logger.info>(candidate.toString());
+ return this.candidateService.updateCandidate(candidate);
+ }
+
+ @PutMapping(value = "/updateRating/{candidateId}&{userId}&{rating}")


stick to REST conventions: put on "/candidates/{candidateId}/rating/" - declare userId and rating as pathvariable, don't use '&' in the url mapping

________________________________

In src/main/java/com/patternpedia/api/rest/controller/IssueController.java<https://github.com/PatternPedia/pattern-pedia-views-backend/pull/6#discussion_r425614384>:

> +import com.patternpedia.api.entities.issue.IssueComment;
+import com.patternpedia.api.entities.issue.Issue;
+import com.patternpedia.api.service.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+//@CrossOrigin(allowedHeaders = "*", origins = "*")
+@RequestMapping(value = "/issue", produces = "application/hal+json")


always in plural: issues/

**Pull request merged**